src/esphomelib/uart_component.cpp

```diff
@@ -22,15 +22,20 @@ UARTComponent::UARTComponent(int8_t tx_pin, int8_t rx_pin, uint32_t baud_rate)
     : tx_pin_(tx_pin), rx_pin_(rx_pin), baud_rate_(baud_rate) {
 #ifdef ARDUINO_ARCH_ESP32
   if (this->rx_pin_ == 3 && this->tx_pin_ == 1) {
-    // Default UART for logging
     this->hw_serial_ = &Serial;
+  } elif (this->rx_pin_ == 9 && this->tx_pin_ == 10) {
+    this->hw_serial_ = &Serial1
+  } elif (this->rx_pin_ == 16 && this->tx_pin_ == 17) {
+    this->hw_serial_ = &Serial2
   } else {
     this->hw_serial_ = new HardwareSerial(next_uart_num++);
   }
 #endif
 #ifdef ARDUINO_ARCH_ESP8266
   if (this->rx_pin_ == 3 && this->tx_pin_ == 1) {
     this->hw_serial_ = &Serial;
+  } elif (this->rx_pin_ == 8 && this->tx_pin_ == 2) {
```

Comment:
It looks like the ESP8266 does not support RX on the UART1 interface (see https://github.com/esp8266/Arduino/blob/master/cores/esp8266/uart.cpp#L631-L637).

Reply:
UART1 supports RX, it's just that the pin (IO8) is also used for SDIO_DATA1 and is disabled when SPI is in use - which is basically always.

UART0 is also interesting in that it supports RTS/CTS on IO13/IO12, and RTS/CTS and RX/TX can be swapped. It's not anywhere near the flexibility of ESP32, but it's not bad given the overall low pinout of the device.